db-developments/python/aggregate.py

```diff
@@ -0,0 +1,105 @@
+from jinja2 import Template
+from datetime import datetime
+from pathlib import Path
+import os
+import sys
+from sqlalchemy import text
+from utils import engine
+
+geoms = {
+    "block": {
+        "source_column": "bctcb{{decade}}",
+        "output_column": "bctcb{{decade}}",
+        "additional_columns": [
+            ("geoid", "cenblock{{decade[:-2]}}")
+        ],
```

Comment:
`additional_columns` seems to always be a list with one single tuple item in it. since it's use in the sql query is:
```sql
{%- for column in additional_columns %}
```
does that end up happy as:
```sql
{%- for column in ("geoid", "cenblock20") %}
```
for example?

Reply:
Ah. No, each tuple refers to one column which both has a source column name and then desired name. So

```
SELECT
{%- for column in [("geoid", "cenblock20")] %}
    {{ column[0] }} AS {{ column[1] }},
{% endfor %}
```
Becomes
```
SELECT
    geoid AS cenblock20,

Reply:
ooo I see, and every other use of it is like
```sql
{%- for column in additional_columns %} 
        {{ column[1] }},
```